src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -722,29 +722,33 @@ internal NullableContextState GetNullableContextState(int position)
 
         internal bool IsGeneratedCode(SyntaxTreeOptionsProvider? provider, CancellationToken cancellationToken)
         {
-            return provider?.IsGenerated(this, cancellationToken) ?? isGeneratedHeuristic();
+            return provider?.IsGenerated(this, cancellationToken) switch
+            {
+                null or GeneratedKind.Unknown => isGeneratedHeuristic(),
+                GeneratedKind kind => kind == GeneratedKind.MarkedGenerated
```

Comment:
```suggestion
                GeneratedKind kind => kind != GeneratedKind.NotGenerated
```

Because we expect others, right?